hub/core/chunk_engine.py

```diff
@@ -363,18 +368,18 @@ def extend(self, samples):
         if len(samples) == 0:
             return
         self._write_initialization()
+        initial_autoflush = self.cache.autoflush
+        self.cache.autoflush = False
+
         samples = self._sanitize_samples(samples)
         indexes_added = get_sample_indexes_added(self.num_samples, samples)
-
         current_chunk = self.last_chunk() or self._create_new_chunk()
-        updated_chunks = {current_chunk}
```

Comment:
instead of maintaining a set of updated chunks, we add every chunk to dirty keys as it is created and also the last_chunk is added to dirty keys if it wasn't present there already.